subprojects/native/src/main/java/org/gradle/internal/nativeintegration/filesystem/services/JdkFallbackHelper.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.nativeintegration.filesystem.services;
+
+import org.gradle.api.JavaVersion;
+import org.gradle.internal.Cast;
+import org.gradle.internal.UncheckedException;
+
+public class JdkFallbackHelper {
+    private JdkFallbackHelper() {
+    }
```

Comment:
Did we start to do this yet?

Reply:
Maybe not. So we are not using private constructors for utility classes?

Reply:
I don't know. I feel like they are just useless boilerplate, especially for internal stuff. If we start doing it, I think we should be consistent, and add it in other places, but first we should have a good reason to start doing it. I don't remember any issues in the past that could have been prevented if we had private constructors for static utilities, so I'm not very keen on changing what we do today.